File: Code/cpp/insertion_sort.cpp
#include<iostream>
using namespace std;
int main()
{
    int length, temp, i, j, select;

    cout << "Enter Length: ";
    cin >> length;
    int arr[length];
    for (i = 0; i < length; i++)
    {
        cout << "Enter Number: ";
        cin >> arr[i];
    }

    for (i = 1; i < length; i++)
    {
        select = i;
        for (j = select - 1; j > -1; j--)
        {
            if (arr[select] < arr[j])
            {
                temp = arr[j];
                arr[j] = arr[select];
                arr[select] = temp;
                select = j;
            }
        }
    }
    for (i = 0; i < length; i++)
        cout << arr[i] << " ";
    return 0;
}

File: Code/py/insertion_sort.py
arr = []
length = int(input("Enter Length: "))

for i in range(length):
    arr.append(int(input("Enter Number: ")))

for i in range(1, length):
    for j in range(i - 1, -1, -1):
        if arr[i] < arr[j]:
            arr[i], arr[j] = arr[j], arr[i]
            i = j
print(arr)